pkg/compactor/partition_visit_marker.go

```diff
@@ -0,0 +1,96 @@
+package compactor
+
+import (
+	"fmt"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// PartitionVisitMarkerDirectory is the name of directory where all visit markers are saved.
+	PartitionVisitMarkerDirectory = "visit-marks"
+	// PartitionVisitMarkerFileSuffix is the known suffix of json filename for representing the most recent compactor visit.
+	PartitionVisitMarkerFileSuffix = "visit-mark.json"
+	// PartitionVisitMarkerFilePrefix is the known prefix of json filename for representing the most recent compactor visit.
+	PartitionVisitMarkerFilePrefix = "partition-"
+	// PartitionVisitMarkerVersion1 is the current supported version of visit-mark file.
+	PartitionVisitMarkerVersion1 = 1
+)
+
+var (
+	errorNotPartitionVisitMarker = errors.New("file is not partition visit marker")
+)
+
+type PartitionVisitMarker struct {
```

Comment:
Can we make all these non public? Do we plan to use them outside of the package?